### DISCUSSION TITLE
problem with using Nub library in py5

### DISCUSSION THREAD

**Discussion by Maltiec** [py5coding/py5generator#80] [Q&A]
hello!

  I've added Nub library to work with 3D, everything seems to be ok (basic example works fine), but then I've tried to use a demo-project and stepped on a problem:

   node.setBehavior(scene, this::behavior);

and this method waits this from me: 

/**
   * Same as {@code graph.addBehavior(this, behavior)}.
   *
   * @see #setBehavior(Graph, Consumer)
   * @see Graph#addBehavior(Node, BiConsumer)
   * @see Graph#addBehavior(Node, Consumer)
   */
  public void setBehavior(Graph graph, BiConsumer<Graph, Node> behavior) {
    graph.addBehavior(this, behavior);
  }

is there any jpype way to correctly use java BiConsumer function?

**Comment by hx2A** [comment]
It's a little hard to understand what problem you are experiencing here. Are you getting an exception? Is it hanging? Sometimes jpype needs some customization to work with object types and the conversion from a Python object to a Java one. Also, Java supports multiple functions with the same name and different signatures, and jpype sometimes can't figure out which one you want and will throw an exception. You can use JConversion to help in these situations.

https://jpype.readthedocs.io/en/latest/userguide.html#type-conversion-customizers
